lib/crawly/bench.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Crawly.Bench do
+  @moduledoc """
+  This module implements a simple benchmarking suite that spawns a local HTTP
+  server to exercise Crawly to the maximum possible speed
+  """
+  require Logger
+
+  alias Crawly.Engine
+
+  @spider_name Crawly.Bench.BenchSpider
+
+  @spec start_benchmark(atom) :: nil
+  def start_benchmark(spider_name) do
+    {:ok, _} = Application.ensure_all_started(:crawly)
+    :ok = Application.put_env(:crawly, :bench, true, persistent: true)
+    Crawly.Engine.start_spider(spider_name)
+    wait_until(spider_name, {0, 0, 0, 0})
+  end
+
+  @spec start_benchmark :: nil
+  def start_benchmark, do: start_benchmark(@spider_name)
+
+  defp wait_until(name, reduc_num_of_items, retries \\ 100, interval \\ 1000)
+
+  defp wait_until(spider_name, {_, _, items, workers}, 0, _) do
+    Logger.info("Stopping #{inspect(spider_name)}")
+    Engine.stop_spider(spider_name, :normal)
+    Logger.info("Max of #{items} requests/sec with #{workers} workers")
+  end
+
+  defp wait_until(
+         name,
+         {reduc, last_num_requests, acc_req, _num_of_workers},
+         retries,
+         interval
+       ) do
+    Process.sleep(interval)
+
+    if Process.whereis(name) do
+      spiders = Engine.running_spiders()
+
+      {:stored_requests, req_count} = Crawly.RequestsStorage.stats(name)
+
+      {_, pid, :worker, _} =
+        Supervisor.which_children(Map.get(spiders, name))
+        |> Enum.find(&({Crawly.Manager, _, :worker, [Crawly.Manager]} = &1))
+
+      {:info, info} = GenServer.call(pid, :collect_metrics)
```

Comment:
This could be an API of CrawlyManager as well. Lets extend it as a separate PR please

Reply:
https://github.com/oltarasenko/crawly/pull/118